### ISSUE TITLE
[3.3.5] Core/Waypoint: Waypoint pause via gossip hello mustn't be executed during escorts

### ISSUE THREAD

**Issue by Funkeln** [TrinityCore/TrinityCore#21190]
**Description:**

Some NPCs wait very long before starting their waypoints during escorts (i.e Arthas @ CoS and Thrall @ Old hillsbrad.

**Current behaviour:**

For example, after talking to Arthas to start the waves (after he moves into the city) it takes full 3 minutes until he starts doing anythiing. 

**Expected behaviour:**

It should not be like that, there should be a small pause when waypoints start, but atm its absurdly high. It makes achievements like the timed on in CoS impossible because waypoint takes too long.

**Steps to reproduce the problem:**

1. Go to CoS
2. Play through the lore yadda-yadda stuff until Arthas move into the city
3. Select his gossip to start waves
4. Notice as you wait forever for him to do anything

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

https://github.com/TrinityCore/TrinityCore/commit/e0b609a178528995959e5ea54255a3f856e62913

**Comment by ccrs** [comment]
most likely previously the timers set by the scripts were just ignored/not honored

**Comment by tje3d** [comment]
Waves start after about 1 or 2 minute, that wait forever is another bug.

**Comment by Aokromes** [comment]
https://github.com/TrinityCore/TrinityCore/pull/17860

**Comment by Funkeln** [comment]
That stratholme rewrite will not fix this. I checked all of his timers, both in his script and script_waypoint. None of them are ever over 20 seconds. This 2 minute wait is coming from elsewhere, something in the core processing the waypoint movement wait timers went wrong.

**Comment by tje3d** [comment]
@Funkeln its not done yet

**Comment by Nyeriah** [comment]
@ccrs The issue is that when a creature has a waypoint sequence in progress, you pause them whenever they are talked to with CONFIG_CREATURE_STOP_FOR_PLAYER (Creature.MovingStopTimeForPlayer - defaulted at 3 minutes) whenever they are talked to. So, Arthas, Thrall and any other escort NPCs that have gossips mid script will have that 3 minutes pause sent OnGossipHello (whenever they are talked to).

There is another issue, however. Even if we lower that setting, and I did try lowering it to 10 seconds, it is still broken. Because then the waypoint is forced to unpause once that setting timer expires, completely ignoring the script. So simply talking to Arthas, without selecting the gossip, still forces him to start moving.

This setting definitely should not apply to creatures scripted with EscortAI. I don't know if there are any exceptions, but definitely not when their script is running.

**Comment by Killyana** [comment]
The pause on gossip hello must be executed only if the waypoint in from `waypoint_data` table, if it's from `waypoints` or `script_waypoint` no pause must be done, it was the behavior before the movement rewrite.

**Comment by jackpoz** [comment]
@ccrs could you take a look at this ?

**Comment by joshwhedon** [comment]
We can reuse the method IsEscortNPC in HandleQuestgiverHelloOpcode and HandleGossipHelloOpcode to check if it is an escort or not, and decide if it can be paused or not. However, I noticed it does not seem to be implemented for smart ai, only for standard c++ escorts. Can extend it by checking the escort state and that the escorter is a player I guess.

**Comment by jackpoz** [comment]
> We can reuse the method IsEscortNPC in HandleQuestgiverHelloOpcode and HandleGossipHelloOpcode

I didn't see any IsEscortNPC method, which line are you referring to ?

**Comment by jackpoz** [comment]
it seems Arthas @ CoS was fixed in the CoS instance rewrite, while Thrall @ Old hillsbrad most likely still has the issue

**Comment by Rushor** [comment]
this has be fixed with https://github.com/TrinityCore/TrinityCore/pull/24005/files ?

**Comment by Killyana** [comment]
No it just fix a double pause

**Comment by jackpoz** [comment]
Arthas has been fixed, Thrall has been fixed. Open a new issue with whatever is broken still
